### PR TITLE
Update DAG example links in multiple providers documents

### LINKED ISSUE #44012: Broken link to docker provider example dags
### What do you see as an issue?

There is a broken link in the [index of providers-docker](https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-docker/index.rst?plain=1#L52):
```Example DAGs <https://github.com/apache/airflow/tree/providers-docker/|version|/providers/tests/system/docker>```

### Solving the problem

If the example DAGs still exist. The link should point to their location.

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by amirmor1**
closes: #44012 

---

Updated multiple provider documents with new DAG example links.

### CODE REVIEW

**Review by vincbeck — APPROVED** [review]

File: docs/apache-airflow-providers-docker/index.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@
     :maxdepth: 1
     :caption: Resources
 
-    Example DAGs <https://github.com/apache/airflow/tree/providers-docker/|version|/providers/tests/system/docker>
+    Example DAGs <https://github.com/apache/airflow/tree/providers-docker/|version|/tests/system/providers/docker>
```
Comment (eladkal) [inline docs/apache-airflow-providers-docker/index.rst:52]: This seems to be wrong for all providers.
can you apply a fix for all?

**Comment by kaxil** [comment]
Could you please update the PR title? The current one is a bit vague and doesn’t clearly communicate the purpose of the change. I recommend reading [this blog post](https://cbea.ms/git-commit/) for guidance.

In particular, it would be helpful to follow this principle:

> **_A properly formed Git commit subject line should always be able to complete the following sentence: “If applied, this commit will [your subject line here].”_**

Since this entry will appear in the release notes, it’s important that the PR title and description clearly explain the changes and their impact.

File: docs/apache-airflow-providers-docker/index.rst (hunk 1 above)
Comment (amirmor1) [reply docs/apache-airflow-providers-docker/index.rst:52]: Done

File: docs/apache-airflow-providers-airbyte/index.rst [hunk 2]
```diff
@@ -55,7 +55,7 @@
     :maxdepth: 1
     :caption: Resources
 
-    Example DAGs <https://github.com/apache/airflow/tree/providers-airbyte/|version|/providers/tests/system/airbyte>
+    Example DAGs <https://github.com/apache/airflow/tree/providers-airbyte/|version|/tests/system/providers/airbyte>
```
Comment (kaxil) [inline docs/apache-airflow-providers-airbyte/index.rst:58]: There is no such link though `https://github.com/apache/airflow/tree/providers-airbyte/`

Comment (amirmor1) [reply docs/apache-airflow-providers-airbyte/index.rst:58]: This link seems to be working. What am I missing here?
https://github.com/apache/airflow/tree/providers-airbyte/4.0.0/tests/system/providers/airbyte

Comment (kaxil) [reply docs/apache-airflow-providers-airbyte/index.rst:58]: aah my bad

**Comment by potiuk** [comment]
We will likely have to adapt some links in released providers already (and @eladkal -> for sure you'd have to apply it and regenerate the docs for current provider's wave.

**Pull request merged**